src/axolotl/utils/config/models/input/v0_4_1/__init__.py

```
@@ -1700,6 +1703,20 @@ def check_kto_config(cls, data):
 
         return data
 
+    @field_validator("sequence_parallel_degree", mode="before")
+    @classmethod
+    def check_sequence_parallel_config(cls, value, info):
+        if not value:
+            value = 1
+
+        if value > 1:
```

Worth checking here if ring-flash-attn is installed so we fail early?

---

Yeah, I like this thought. I think in general we can do a lot of these import checks (that we're currently doing elsewhere in the codebase) at the pydantic model level.